### PR TITLE
[WalletLink] Cleanup: deprecate internal types, remove single use types, consolidate metadata, etc.

### PR DESCRIPTION
**Pull request by bangtoven**
### _Summary_
image

- deprecate JSONRPCRequest, JSONRPCResponse, JSONRPCMethod 
- simplify call stack 
- remove single-use interfaces
- consolidate metadata
- simplify access control

### _How did you test your changes?_

unit tests

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts [hunk 1]
```diff
@@ -27,52 +25,21 @@ import { fetchRPCRequest } from ':util/provider';
...
-interface WatchAssetParams {
-  type: string;
-  options: {
-    address: string;
-    symbol?: string;
-    decimals?: number;
-    image?: string;
-  };
-}
+type RequestParam = unknown[];
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts:28]: temporary type for further refactor

File: packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts [hunk 2]
```diff
@@ -27,52 +25,21 @@ import { fetchRPCRequest } from ':util/provider';
...
 
 // original source: https://github.com/coinbase/coinbase-wallet-sdk/blob/v3.7.1/packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts
 export class WalletLinkSigner implements Signer {
-  private _appName: string;
-  private _appLogoUrl: string | null;
+  private metadata: AppMetadata;
   private _relay: WalletLinkRelay | null = null;
   private readonly _storage: ScopedLocalStorage;
-  private readonly _relayEventManager: RelayEventManager;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts:63]: unnecessary

File: packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts [hunk 3]
```diff
@@ -118,17 +85,38 @@ export class WalletLinkSigner implements Signer {
...
+    if (!request?.options) {
+      throw standardErrors.rpc.invalidParams('Options are required');
+    }
+
+    if (!request?.options.address) {
+      throw standardErrors.rpc.invalidParams('Address is required');
+    }
+
+    const chainId = this.getChainId();
+    const { address, symbol, image, decimals } = request.options;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts:115]: consolidated to simplify call stack
image

File: packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts [hunk 4]
```diff
@@ -141,30 +129,50 @@ export class WalletLinkSigner implements Signer {
...
-    if (!this._isAuthorized()) {
-      await relay.requestEthereumAccounts();
-    }
+    const {
+      rpcUrls = [],
+      blockExplorerUrls = [],
+      chainName,
+      iconUrls = [],
+      nativeCurrency,
+    } = request;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts:173]: image

File: packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts [hunk 5]
```diff
@@ -175,13 +183,18 @@ export class WalletLinkSigner implements Signer {
...
+    throw standardErrors.rpc.internal('unable to add ethereum chain');
   }
 
-  private async switchEthereumChain(chainId: number) {
+  private async switchEthereumChain(params: RequestParam) {
+    const request = params[0] as {
+      chainId: string;
+    };
+    const chainId = parseInt(request.chainId, 16);
+
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts:197]: image

File: packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts [hunk 6]
```diff
@@ -215,23 +230,7 @@ export class WalletLinkSigner implements Signer {
...
-    // Coinbase Wallet Requests
-    const id = this._relayEventManager.makeRequestId();
-    const result = await this._sendRequestAsync({
-      method,
-      params: newParams,
-      jsonrpc: '2.0',
-      id,
-    });
-
-    return result.result;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts:231]: 🍝

File: packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts [hunk 7]
```diff
@@ -247,53 +246,20 @@ export class WalletLinkSigner implements Signer {
...
-
-      default:
-        return undefined;
-    }
-  }
-
-  private async _handleAsynchronousMethods(request: JSONRPCRequest): Promise<JSONRPCResponse> {
-    const { method } = request;
-    const params = request.params || [];
+        return hexStringFromNumber(this.getChainId());
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts:261]: image

File: packages/wallet-sdk/src/sign/walletlink/relay/type/JSONRPC.ts [hunk 8]
```diff
@@ -1,58 +0,0 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
-// Copyright (c) 2018-2023 Coinbase, Inc. <https://www.coinbase.com/>
-
-export type JSONRPCMethod =
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/relay/type/JSONRPC.ts:4]: all BS

Comment (willnewman-cb) [reply packages/wallet-sdk/src/sign/walletlink/relay/type/JSONRPC.ts:4]: 😂

**Review by fan-zhang-sv — APPROVED** [review]
lets go

**Pull request merged**